### PR TITLE
Add Private registry migration

### PR DESCRIPTION
**Pull request by galal-hussein**
1- The PR Adds the following flag:

--registry: string slice flag that specifies the TLS configuration for registries, the syntax of the flag value must be as follows:

`<registry url>,<ca cert path>,<cert path>,<key path>`

The agent will loop over these values and configure the registries.yaml file used by containerd, since there is no method to specify the TLS certs for registries in the cluster.yml file for rke1

2- The pr will allow the agent to copy the cni plugin name used by rke1

https://github.com/rancher/migration-agent/issues/6

### CODE REVIEW

**Review by brandond — APPROVED** [review]

**Review by dereknola — APPROVED** [review]

File: pkg/config/config.go [hunk 1]
```diff
@@ -162,3 +151,67 @@ func ExportKubeProxyConfig(fullState *cluster.FullState, dataDir string) error {
...
+	if _, err := os.Stat(privateRegistryPath); err != nil {
+		if os.IsNotExist(err) {
+			r := registries.Registry{}
+			r.Configs = make(map[string]registries.RegistryConfig)
+			for _, reg := range privateRegistryConfig {
+				endpoint := reg.URL
+				if endpoint == "" {
+					endpoint = "docker.io"
+				}
+				if reg.User != "" && reg.Password != "" {
```
Comment (briandowns) [inline pkg/config/config.go:169]: Could there ever be a circumstance where a user is set and no password?

File: pkg/config/config.go [hunk 2]
```diff
@@ -162,3 +151,67 @@ func ExportKubeProxyConfig(fullState *cluster.FullState, dataDir string) error {
...
+}
+
+func getRegistryTLSConfig(endpoint string, registriesTLS []string) *registries.TLSConfig {
+	if len(registriesTLS) <= 0 {
+		return nil
+	}
+	var caCert, cert, key, url string
+	for _, registryTLS := range registriesTLS {
+		certs := strings.Split(registryTLS, ",")
+		if len(certs) < 4 {
```
Comment (briandowns) [inline pkg/config/config.go:200]: Can you make 4 a const with a descriptive name?

File: pkg/config/config.go (hunk 1 above)
Comment (galal-hussein) [reply pkg/config/config.go:169]: I dont believe thats possible with basic authentication for private registry

File: pkg/config/config.go (hunk 2 above)
Comment (galal-hussein) [reply pkg/config/config.go:200]: will fix

File: pkg/config/config.go [hunk 3]
```diff
@@ -162,3 +152,67 @@ func ExportKubeProxyConfig(fullState *cluster.FullState, dataDir string) error {
...
+
+	return nil
+}
+
+func getRegistryTLSConfig(endpoint string, registriesTLS []string) *registries.TLSConfig {
+	if len(registriesTLS) <= 0 {
+		return nil
+	}
+	var caCert, cert, key, url string
+	for _, registryTLS := range registriesTLS {
```
Comment (briandowns) [inline pkg/config/config.go:199]: Can we safely assume that the URLs passed in here are all valid? If not, should we first validate them before splitting them into parts?

**Review by briandowns — APPROVED** [review]
LGTM

**Pull request merged**